tests/functional/adapter/constraints/test_constraints.py

```diff
@@ -0,0 +1,213 @@
+
+    @pytest.fixture(scope="class")
+    def expected_sql(self):
+        return _expected_sql_trino
+
+
+class TestTrinoTableConstraintsRollback(BaseConstraintsRollback):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_sql,
+            "constraints_schema.yml": trino_model_schema_yml,
+        }
+
+    # We are trying to load char value to integer column to break constraint.
+    # In BaseConstraintsRollback it is checked by violating not-null constraint,
+    # But not-null constraint is not supported in dbt-trino
+    @pytest.fixture(scope="class")
+    def null_model_sql(self):
+        return trino_model_char_value_to_int_column
+
+    @pytest.fixture(scope="class")
+    def expected_error_messages(self):
+        return [
+            "Please ensure the name, data_type, and number of columns in your contract match the columns in your model's definition."
+        ]
+
+
+class TestTrinoIncrementalConstraintsRuntimeDdlEnforcement(
+    BaseIncrementalConstraintsRuntimeDdlEnforcement
+):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_incremental_wrong_order_sql,
+            "constraints_schema.yml": trino_model_schema_yml,
+        }
+
+    @pytest.fixture(scope="class")
+    def expected_sql(self):
+        return _expected_sql_trino
+
+
+class TestTrinoIncrementalConstraintsRollback(BaseIncrementalConstraintsRollback):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_incremental_model_sql,
+            "constraints_schema.yml": trino_model_schema_yml,
+        }
+
+    # We are trying to load char value to integer column to break constraint.
+    # In BaseConstraintsRollback it is checked by violating not-null constraint,
+    # But not-null constraint is not supported in dbt-trino
+    @pytest.fixture(scope="class")
+    def null_model_sql(self):
+        return trino_model_char_value_to_int_column
+
+    @pytest.fixture(scope="class")
+    def expected_error_messages(self):
+        return [
+            "Please ensure the name, data_type, and number of columns in your contract match the columns in your model's definition."
+        ]
+
+
+class TestTrinoModelConstraintsRuntimeEnforcement(BaseModelConstraintsRuntimeEnforcement):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_sql,
+            "constraints_schema.yml": trino_constrained_model_schema_yml,
+        }
+
+    @pytest.fixture(scope="class")
+    def expected_sql(self):
+        return """
+create table <model_identifier> (
+    id integer,
+    color varchar,
+    date_day varchar
+) ;
+insert into <model_identifier>
+(
+    select
+        id,
+        color,
+        date_day from
+    (
+        select
+            1 as id,
+            'blue' as color,
+            '2019-01-01' as date_day
+    ) as model_subq
+)
```

Comment:
This could be written as a single query like this:
```
create table <model_identifier>
as
select 
    cast('blue' as varchar) as color,
    cast(1 as integer) as id,
    cast('2019-01-01' as date) as date_day
```